File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/investigation/mapsmenu/mapdisplay/io/MapFileIO.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.mapsmenu.mapdisplay.io;

import androidx.annotation.NonNull;

import com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.mapsmenu.mapdisplay.io.models.MapDesBlueprint;

import java.io.InputStream;

public class MapFileIO {

    public final MapDesBlueprint deserializedData = new MapDesBlueprint();

    public final MapFileReader reader = new MapFileReader(deserializedData);

    /** @noinspection UnusedReturnValue*/
    public boolean readFile(InputStream inputStream, @NonNull MapFileReader reader) {
        return reader.loadFile(inputStream);
    }

}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/investigation/mapsmenu/mapdisplay/io/MapFileIO.kt
package com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.mapsmenu.mapdisplay.io

import com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.mapsmenu.mapdisplay.io.models.MapDesBlueprint
import java.io.InputStream

class MapFileIO {

    private val deserializedData: MapDesBlueprint = MapDesBlueprint()

    val reader: MapFileReader = MapFileReader(deserializedData)

    fun readFile(inputStream: InputStream?, reader: MapFileReader): Boolean {
        return reader.loadFile(inputStream)
    }
}
